src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -700,12 +707,12 @@ void queue_waiting_task(LocalSchedulerState *state,
   TaskQueueEntry task_entry = TaskQueueEntry_init(spec, task_spec_size);
   queue_task(state, algorithm_state->waiting_task_queue, &task_entry,
              from_global_scheduler);
-  /* If we're queueing this task in the waiting queue, there must be at least
-   * one missing dependency, so record it. The TaskSpec pointer that is added
-   * to the missing dependencies' dependent tasks must match the TaskSpec
-   * pointer added to the waiting queue, so that we can later remove the
-   * correct queue entry. */
-  fetch_missing_dependencies(state, algorithm_state, &task_entry);
+  /* There must be at least one missing dependency, so record it. The task was
+   * just queued, so we can get a reference to it by going to the last element
+   * in the waiting queue. */
```

Comment:
We have to get the iterator somehow, and I think this is the best way to do it. I changed this so that `queue_task` returns the correct iterator, though, so we can be sure that no one modified the list in between.